Writing your own application (#186)
esp-rs/book: src/writing-your-own-application/async.md

# Async

This subchapter does not serve as an `async` tutorial or teaching material. For these purposes visit [official async-book]. 

> 

Reviewer: We should at least say something like `embassy` is supported by esp-hal ... just starting to talk about embassy looks a bit odd otherwise

While the next paragraph explains it, just starting to explain what `embassy` is feels weird

Author: I re-order the paragraph a bit, does it look better or does it need improvement?

# Async

This subchapter does not serve as an `async` tutorial or teaching material. For these purposes visit [official async-book]. 

The [esp-hal-embassy] crate provides integration between the [esp-hal] and the [Embassy] asynchronous framework. It provides support for:

1. Interrupt-mode executor
2. Multicore-aware thread-mode embassy executor
3. Embassy time driver
4. Timer waiter queue

`esp-hal` provides `blocking` and `async` API for most of the supported drivers. For more information and to get started check our `examples` in the [esp-hal-package].

## Embassy

> [Embassy] is an asynchronous (async) framework designed specifically for embedded Rust development which consists of several crates that you can use together or independently. One of the projects is [embassy-executor]. The `embassy-executor` is an `async/await` executor that generally executes a fixed number of tasks, allocated at startup, though more can be added later. For more information about `embassy` visit [Embassy book].

Reviewer 2: ```suggestion
[Embassy] is an asynchronous (async) framework designed specifically for embedded Rust development; it [embassy-executor] crate provides an `async/await` executor which executes a fixed number of tasks, statically allocated at startup, though more can be added later. For more information about `embassy` visit [Embassy book].
```
Feel free to reword further, but something like this sounds a bit more natural to me.

## RTIC

> [Real-Time Interrupt-driven Concurrency (RTIC)] is a concurrency framework for building real-time systems. Currently, only the ESP32-C3 is supported.

Reviewer 2: There is an open PR for C6 support, so if we aren't going to mention this now we should probably at least open an issue so that we don't forget to add it here when support has been merged.

Author: I added C6 mentioned
